app/assets/javascripts/discourse/app/services/interface-color.js

```diff
@@ -10,6 +11,56 @@ export default class InterfaceColor extends Service {
   @service siteSettings;
   @service session;
   @tracked forcedColorMode;
+  @tracked browserPrefersDark = this.#browserPreferenceMatcher.matches;
+
+  #browserPreferenceMatcher = window.matchMedia("(prefers-color-scheme: dark)");
+
+  constructor() {
+    super(...arguments);
+    this.#browserPreferenceMatcher.addEventListener(
+      "change",
+      this._handleBrowserPreferenceChange
+    );
+  }
+
+  willDestroy() {
+    this.#browserPreferenceMatcher.removeEventListener(
+      "change",
+      this._handleBrowserPreferenceChange
+    );
+  }
+
+  get darkMediaQuery() {
+    if (!this.session.darkModeAvailable) {
+      return "none";
+    } else if (this.session.defaultColorSchemeIsDark) {
```

Comment:
I think `defaultColorSchemeIsDark` should be checked before `darkModeAvailable` since there could be a situation where a site's default palette is considered dark (per this [logic](https://github.com/discourse/discourse/blob/834ea70b1ce3f06e6e03d25ebee0567596ad26c2/app/models/color_scheme.rb#L521-L528)), and there are no other color palettes that users can select which would result in the `darkModeAvailable` property becoming false. In this scenario, we'd want the dark media query to be "all", but that wouldn't happen if we check `darkModeAvailable` first.